### DISCUSSION TITLE
structuredClone is not defined

### DISCUSSION THREAD

**Discussion by aksyuma** [renovatebot/renovate#22631] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.110.1

### If you're self-hosting Renovate, select which platform you are using.

AWS CodeCommit

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

Renovate fails to execute as described below..

To reproduce it do following:

1.  Create an AWS Codecommit repository with some dependency
2.  Install renovate using command :  `npm install -g renovate`
3.  Add the `renovate.json` with the following content:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "enabled": true,
  "schedule": ["at any time"]
  "packageRules": [
    {
      "updateTypes": ["minor", "patch", "pin", "digest"],
      "automerge": true
    }
  ]
}
```
4. Execute: `renovate` and notice the following exception :

```
FATAL: Fatal error: structuredClone is not defined
       "err": {
         "message": "structuredClone is not defined",
         "stack": "ReferenceError: structuredClone is not defined\n    at mergeChildConfig (/usr/local/lib/node_modules/renovate/lib/config/utils.ts:14:24)\n    at parseConfigs (/usr/local/lib/node_modules/renovate/lib/workers/global/config/parse/index.ts:27:43)\n    at /usr/local/lib/node_modules/renovate/lib/workers/global/index.ts:125:16\n    at start (/usr/local/lib/node_modules/renovate/lib/workers/global/index.ts:123:5)\n    at /usr/local/lib/node_modules/renovate/lib/renovate.ts:18:22"
       }
```

Alternative Suggestions already tried:

1.  Execute the `renovate-config-validator`

and notice the following exception :

```
FATAL: Fatal error: structuredClone is not defined
       "err": {
         "message": "structuredClone is not defined",
         "stack": "ReferenceError: structuredClone is not defined\n    at mergeChildConfig (/usr/local/lib/node_modules/renovate/lib/config/utils.ts:14:24)\n    at parseConfigs (/usr/local/lib/node_modules/renovate/lib/workers/global/config/parse/index.ts:27:43)\n    at /usr/local/lib/node_modules/renovate/lib/workers/global/index.ts:125:16\n    at start (/usr/local/lib/node_modules/renovate/lib/workers/global/index.ts:123:5)\n    at /usr/local/lib/node_modules/renovate/lib/renovate.ts:18:22"
       }
```

2. Created global `config.js` in the root of the repository
2.1. Execute: `renovate` and notice the following exception :

```
FATAL: Fatal error: structuredClone is not defined
       "err": {
         "message": "structuredClone is not defined",
         "stack": "ReferenceError: structuredClone is not defined\n    at mergeChildConfig (/usr/local/lib/node_modules/renovate/lib/config/utils.ts:14:24)\n    at parseConfigs (/usr/local/lib/node_modules/renovate/lib/workers/global/config/parse/index.ts:27:43)\n    at /usr/local/lib/node_modules/renovate/lib/workers/global/index.ts:125:16\n    at start (/usr/local/lib/node_modules/renovate/lib/workers/global/index.ts:123:5)\n    at /usr/local/lib/node_modules/renovate/lib/renovate.ts:18:22"
       }
```

## Local development Env
OS : Ubuntu - 18.04
nvm : 0.39.2
Renovate: v35.110.1

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>
FATAL: Fatal error: structuredClone is not defined
       "err": {
         "message": "structuredClone is not defined",
         "stack": "ReferenceError: structuredClone is not defined\n    at mergeChildConfig (/usr/local/lib/node_modules/renovate/lib/config/utils.ts:14:24)\n    at parseConfigs (/usr/local/lib/node_modules/renovate/lib/workers/global/config/parse/index.ts:27:43)\n    at /usr/local/lib/node_modules/renovate/lib/workers/global/index.ts:125:16\n    at start (/usr/local/lib/node_modules/renovate/lib/workers/global/index.ts:123:5)\n    at /usr/local/lib/node_modules/renovate/lib/renovate.ts:18:22"
       }

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment, marked as answer]
Which version of node are you running? You need node 18

**Reply by aksyuma** [reply]
I'm using node v16. I will update to 18 as per recommendation and report back soonest...

**Reply by aksyuma** [reply]
> Which version of node are you running? You need node 18

upgrading node from `v16` to `v18.16.0` seems to have resolved the error: `structuredClone is not defined` . However I am getting a follow-up exception after running: `renovate`

```
 INFO: Repository started (repository=**REDACTED**)
       "renovateVersion": "35.114.3"
 INFO: Repository has invalid config (repository=**REDACTED**)
       "error": {
         "validationSource": "renovate.json",
         "validationError": "Invalid JSON (parsing failed)",
         "validationMessage": "Syntax error: expecting end of expression or separator near \"]\n  \"pack",
         "message": "config-validation",
         "message": "config-validation",
         "stack": "Error: config-validation\n    at checkForRepoConfigError (/home/ubuntu/.nvm/versions/node/v18.16.0/lib/node_modules/renovate/lib/workers/repository/init/merge.ts:181:17)\n    at mergeRenovateConfig (/home/ubuntu/.nvm/versions/node/v18.16.0/lib/node_modules/renovate/lib/workers/repository/init/merge.ts:205:3)\n    at getRepoConfig (/home/ubuntu/.nvm/versions/node/v18.16.0/lib/node_modules/renovate/lib/workers/repository/init/config.ts:12:12)\n    at initRepo (/home/ubuntu/.nvm/versions/node/v18.16.0/lib/node_modules/renovate/lib/workers/repository/init/index.ts:43:12)\n    at Object.renovateRepository (/home/ubuntu/.nvm/versions/node/v18.16.0/lib/node_modules/renovate/lib/workers/repository/index.ts:54:14)\n    at attributes.repository (/home/ubuntu/.nvm/versions/node/v18.16.0/lib/node_modules/renovate/lib/workers/global/index.ts:184:11)\n    at start (/home/ubuntu/.nvm/versions/node/v18.16.0/lib/node_modules/renovate/lib/workers/global/index.ts:169:7)\n    at /home/ubuntu/.nvm/versions/node/v18.16.0/lib/node_modules/renovate/lib/renovate.ts:18:22"
       }
 INFO: Repository finished (repository=**REDACTED**)
       "cloned": true,
       "durationMs": 9134
```
Interesting enough `renovate.json` has been validated using `renovate-config-validator` as shown below:

```
renovate-config-validator
 INFO: Validating renovate.json
 INFO: Validating config.js
 INFO: Config validated successfully
```

**Reply by rarkins** [reply]
This topic of this discussion is answered, please don't take it off topic
